### PR TITLE
Use gwapiv1a2.PolicyStatus for SecurityPolicy Status

### PR DESCRIPTION
**Pull request by zhaohuabing**
This is part of https://github.com/envoyproxy/gateway/issues/2631.

It's better to address this before GA since it's an API change.

### CODE REVIEW

File: internal/gatewayapi/securitypolicy.go [hunk 1]
```diff
@@ -105,65 +111,126 @@ func (t *Translator) ProcessSecurityPolicies(securityPolicies []*egv1a1.Security
...
+						Name:        p.Name,
+						SectionName: p.SectionName,
+					})
 				}
 			}
 
-			err := t.translateSecurityPolicyForRoute(policy, route, resources, xdsIR)
 			if err != nil {
-				status.SetSecurityPolicyCondition(policy,
+				status.SetSecurityPolicyCondition(
```
Comment (shawnh2) [inline internal/gatewayapi/securitypolicy.go:126]: hi @zhaohuabing, can this be merged after #2846, since it introduces a more generic method for `PolicyStatus`?

Comment (zhaohuabing) [reply internal/gatewayapi/securitypolicy.go:126]: Sure, I'll change this PR to draft and revisit it after 2846.

**Comment by arkodg** [comment]
hey @shawnh2 can you fix the conflict ?

File: internal/gatewayapi/securitypolicy.go [hunk 2]
```diff
@@ -80,98 +80,157 @@ func (t *Translator) ProcessSecurityPolicies(securityPolicies []*egv1a1.Security
...
+			}
+
+			err := validatePortOverlapForSecurityPolicyRoute(xdsIR, targetedRoute)
 			if err == nil {
-				err = t.translateSecurityPolicyForRoute(policy, route, resources, xdsIR)
+				err = t.translateSecurityPolicyForRoute(policy, targetedRoute, resources, xdsIR)
 			}
+			if err == nil {
+				err = t.translateSecurityPolicyForRoute(policy, targetedRoute, resources, xdsIR)
+			}
```
Comment (shawnh2) [inline internal/gatewayapi/securitypolicy.go:143]:
```suggestion
```

**Review by shawnh2 — APPROVED** [review]
LGTM, one small nit

File: internal/provider/kubernetes/controller.go [hunk 3]
```diff
@@ -839,9 +839,8 @@ func (r *gatewayAPIReconciler) processSecurityPolicies(
 
 	for _, policy := range securityPolicies.Items {
 		policy := policy
-		// Discard Status to reduce memory consumption in watchable
-		// It will be recomputed by the gateway-api layer
-		policy.Status = egv1a1.SecurityPolicyStatus{}
+		// The status of BackendTLSPolicies can't be discarded because the status
```
Comment (shawnh2) [inline internal/provider/kubernetes/controller.go:842]: seems we can revert this change according to #2902

File: internal/gatewayapi/testdata/securitypolicy-no-status-for-unknown-gateway-or-route.out.yaml [hunk 4]
```diff
@@ -12,7 +12,8 @@ securityPolicies:
       kind: HTTPRoute
       name: unknown-httproute
       namespace: envoy-gateway
-  status: {}
```
Comment (arkodg) [inline internal/gatewayapi/testdata/securitypolicy-no-status-for-unknown-gateway-or-route.out.yaml:15]: we need empty status

Comment (arkodg) [reply internal/gatewayapi/testdata/securitypolicy-no-status-for-unknown-gateway-or-route.out.yaml:15]: nvm this this empty

**Review by shawnh2 — APPROVED** [review]
